versioned/persist/adapter/src/main/java/org/projectnessie/versioned/persist/adapter/DatabaseAdapter.java

```diff
@@ -150,7 +152,10 @@ Hash commit(CommitAttempt commitAttempt)
    *     expected hEAD
    */
   Hash transplant(
-      BranchName targetBranch, Optional<Hash> expectedHead, List<Hash> sequenceToTransplant)
+      BranchName targetBranch,
+      Optional<Hash> expectedHead,
+      List<Hash> sequenceToTransplant,
+      Function<ByteString, ByteString> resetWithMergeProps)
```

Comment:
Can you replace `resetWithMergeProps` here and elsewhere with `updateCommitMetadata`?